File: packages/3/ms-3-flaskpy/flaskr/routes/v1/country/country_routes.py
from flask import Blueprint, jsonify, request
from flaskr.middlewares.validator import validator_middleware
from .country_controller import CountryController

country_blueprint = Blueprint('country', __name__, url_prefix='/country')


@country_blueprint.route('/create/one', methods=['POST'])
def create_one():
    country = CountryController()
    return jsonify(country.create_one_country(request.json))


@country_blueprint.route('/update/one', methods=['PUT'])
def update_one():
    print(request.json)
    country = CountryController()
    return jsonify(country.update_one_country(request.json))


@country_blueprint.route('/delete/one', methods=['DELETE'])
def delete_one():
    country = CountryController()
    return jsonify(country.delete_one_country(request.json))


@country_blueprint.route('/search/one', methods=['POST'])
def search_one():
    country = CountryController()
    return jsonify(country.search_one_country(request.json))


@country_blueprint.route('/search/many', methods=['POST'])
@validator_middleware
def search_many():
    country = CountryController()
    return jsonify(country.search_many_country(request.json))

File: packages/3/ms-3-expressjs/src/routes/v1/country.routes.ts
import { iresponse, createOneCountrySchema, updateOneCountrySchema, deleteOneCountrySchema, searchOneCountrySchema, searchManyCountrySchema } from '@gmahechas/erp-common';
import { express, authenticationMiddleware, validatorMiddleware, loggerMiddleware } from '@gmahechas/erp-common-expressjs';
import { createOneCountry, updateOneCountry, deleteOneCountry, searchOneCountry, searchManyCountry } from '@gmahechas/erp-common-ms-3-js';

const router = express.Router();

router.post('/create/one', authenticationMiddleware, validatorMiddleware(createOneCountrySchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await createOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.put('/update/one', authenticationMiddleware, validatorMiddleware(updateOneCountrySchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await updateOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.delete('/delete/one', authenticationMiddleware, validatorMiddleware(deleteOneCountrySchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await deleteOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.post('/search/one', authenticationMiddleware, validatorMiddleware(searchOneCountrySchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await searchOneCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.post('/search/many', authenticationMiddleware, validatorMiddleware(searchManyCountrySchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await searchManyCountry(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

export const countryRouter = router;